File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        countOfOccurs = []
        sets = set(arr)
        # for i in sets:
        #     count = 0
        #     for j in arr:
        #         if i == j:
        #             count += 1
        #     countOfOccurs.append(count)
        for i in sets:
            countOfOccurs.append(arr.count(i))
            
        for i in countOfOccurs:
            if countOfOccurs.count(i) > 1:
                return False  
        return True

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.java
import java.util.*;

class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> freq = new HashMap<>();
        for (int num : arr)
            freq.put(num, freq.getOrDefault(num, 0) + 1);

        Set<Integer> uniqueCounts = new HashSet<>(freq.values());
        return uniqueCounts.size() == freq.size();
    }
}
